db/1.9/modules/ROOT/pages/database-connector-select.adoc

```diff
@@ -0,0 +1,184 @@
+= Query a Database Examples - Mule 4
+:page-aliases: connectors::db/db-connector-query.adoc
+
+Anypoint Connector for Database (Database Connector) Select operation selects data from a database. These examples help you configure the operation in Studio and XML.
+
+== Configure the Select Operation in Studio
+
+To add and configure the Select operation in Studio, follow these steps:
+
+. In the *Mule Palette* view, search for `database` and select the *Select* operation.
+. Drag the *Select* operation onto the Studio canvas.
+. In the *General* tab of the operation configuration screen, click the plus sign (*+*) next to the *Connector configuration* field to access the global element configuration fields.
+. Specify the database connection information and click *OK*.
+. Set the *SQL Query Text* field to the SQL query to execute, for example `SELECT * FROM PLANET WHERE name = :name`.
+
+The following screenshot shows the *Select* operation configuration:
+
+.Select operation configuration
+image::database-select-operation-1.png[Select operation configuration in Studio]
+
+In the XML editor, the `<db:select>` configuration looks like this:
+[source,xml,linenums]
+----
+<db:select config-ref="dbConfig">
+  <db:sql>SELECT * FROM PLANET WHERE name = :name</db:sql>
+</db:select>
+----
+
+== Configure the Input Parameters Field
+
+To protect database queries, configure the *Input Parameters* field in the Select operation by adding variable values to the SQL statement you execute in the database. The primary goal of the Select operation is to supply a SQL query and use DataWeave for the parameters.
+
+The parameters you provide are a map in which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a colon prefix, for example `where id = :myParamName`. The map’s values contains the actual assignation for each parameter.
+
+The advantages of using input parameters to configure the `WHERE` clause in a SQL `SELECT` statement are that it makes the query immune to SQL injection attacks and enables optimizations that are not possible otherwise improving the application performance.
+
+For security reasons, do not directly write `<db:sql>SELECT * FROM PLANET WHERE name = #[payload] </db:sql>`.
+
+DataSense is available for the operation's input and output. Database Connector analyzes the query and automatically calculates the structure of the query's output by analyzing the projection section of the SQL statement. At the same time, by comparing the conditions in the `WHERE` clause to the table structure, it generates DataSense input to help you build the DataWeave script that generates the input parameters.
```

Comment:
DataSense > DataSense Explorer? DataSense Query Language?

Reply:
What does it mean to say that DataSense "is available for input and output"? available to use to create input and output? to receive input and output? Or some other action?